### ISSUE TITLE
Kong 3.6: resty.openssl.auxiliary.nginx doesn't support Nginx version 1025003

### ISSUE THREAD

**Issue by LukeMccon** [Kong/kong#12592]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

3.6.0

### Current Behavior

When I run the Kong 3.6-ubuntu docker image in the hybrid mode, the control plane's migrations crash giving the following error: `resty.openssl.auxiliary.nginx doesn't support Nginx version 1025003`

### Expected Behavior

Migrations run with no error

### Steps To Reproduce

1. Setup kong hybrid mode in docker compose or similar environment
2. use `kong:3.6-ubuntu` docker image
3. Setup PostgreSQL database with TLS enabled
4. Execute `kong migrations --v up ` in control plane container 

### Anything else?

- I was not encountering this issue when the PostgresSQL database was configured locally to *not* use TLS. The TLS connection triggers the issue.
- I followed the upgrade notes and setting `ssl_cipher_suite` to `old` didn't resolve the issue
- With an identical setup using `kong:3.5-ubuntu` the control plane's migrations work prefectly

Versions:
nginx version: openresty/1.25.3.1
kong: 3.6.0

**Comment by bungle** [comment]
@samugi, I think he is hitting this:
https://github.com/Kong/lua-resty-openssl/blob/master/lib/resty/openssl/auxiliary/nginx.lua#L43-L90

**Comment by fffonion** [comment]
Hi @LukeMccon could you share your kong.conf content, the postgres server version (is it higher than 11 and using scram auth?) and if there's any custom plugin that you are using?
And if possible, also the full stack trace for the error you mentioned.

**Comment by marco-sciatta** [comment]
Same issue here

**Comment by duonghanu** [comment]
I got the same issue when upgrade from Kong OSS 3.5.0 to 3.6.0 when run migration db process

kong migrations up --vv
`
Error:
resty.openssl.auxiliary.nginx doesn't support Nginx version 1025003
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/resty/openssl/auxiliary/nginx.lua:89: in main chunk
        [C]: in function 'require'
        /usr/local/share/lua/5.1/resty/openssl/ssl.lua:8: in main chunk
        [C]: in function 'require'
        /usr/local/share/lua/5.1/pgmoon/init.lua:397: in function 'auth'
        /usr/local/share/lua/5.1/pgmoon/init.lua:268: in function 'connect'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:215: in function 'connect'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:546: in function 'query'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:296: in function 'init'
        /usr/local/share/lua/5.1/kong/db/init.lua:144: in function 'init_connector'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:101: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:31: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:31>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:31: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:15>
        (command line -e):7: in function 'inline_gen'
        init_worker_by_lua(nginx.conf:132):44: in function <init_worker_by_lua(nginx.conf:132):43>
        [C]: in function 'xpcall'
        init_worker_by_lua(nginx.conf:132):52: in function <init_worker_by_lua(nginx.conf:132):50>
`

**Comment by chronolaw** [comment]
Hi @LukeMccon , I think that we have fixed this issue in #12665, could you try it again? thanks.

**Comment by Water-Melon** [comment]
@LukeMccon This issue has been fixed in #12905 and will be released in the next version.

**Comment by LukeMccon** [comment]
Thank you @Water-Melon @chronolaw and team, confirmed that this issue is fixed in OSS 3.6.1

**Comment by duonghanu** [comment]
I got the following error when upgrade Kong OSS 3.5.0 -> 3.6.1. I'm using AWS RDS Postgres 15

```
Error:
/usr/local/share/lua/5.1/pgmoon/init.lua:398: attempt to index local 'ssl' (a nil value)
stack traceback:
        /usr/local/share/lua/5.1/pgmoon/init.lua:398: in function 'auth'
        /usr/local/share/lua/5.1/pgmoon/init.lua:268: in function 'connect'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:215: in function 'connect'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:546: in function 'query'
        .../share/lua/5.1/kong/db/strategies/postgres/connector.lua:296: in function 'init'
        /usr/local/share/lua/5.1/kong/db/init.lua:144: in function 'init_connector'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:101: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:31: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:31>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:31: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:15>
        (command line -e):7: in function 'inline_gen'
        init_worker_by_lua(nginx.conf:127):44: in function <init_worker_by_lua(nginx.conf:127):43>
        [C]: in function 'xpcall'
        init_worker_by_lua(nginx.conf:127):52: in function <init_worker_by_lua(nginx.conf:127):50>
```
